### DISCUSSION TITLE
Add customize visualization to runner.test

### DISCUSSION THREAD

**Discussion by HyoKong** [open-mmlab/mmengine#1006] [General]
Hi, I want to visualize the intermediate feature maps for HRNet during the testing step.
```python
runner = Runner.from_cfg(cfg)
visualizer = Visualizer(...)
test_loader = runner.test_dataloader
model = runner.model
model.eval()
register_feature_hook(model)  # register_feature_hook is my function for obtaining intermediate feature maps

for i, data in enumerate(test_loader):
    with torch.no_grad():
        post_data = model.data_preprocessor(data, training=False)
        model(**post_data)
        feats = hidden_outputs['head.final_layer']
        for idx in range(len(data['inputs'])):
            drawn_img = visualizer.draw_featmap(featmap=feats[idx], overlaid_image=data['inputs'][idx].permute(1, 2, 0).data.cpu().numpy(),
                                                channel_reduction='squeeze_mean',
                                                alpha=0.5)
            visualizer.add_image(f'feats{i}', drawn_img, step=idx)
```

May I ask how to integrate the function into runner.test() cos I want to know the accuracy metrics. Thank you so much!
